src/backend/TrafficCourts/Common/Features/FilePersistence/MinioFilePersistenceService.cs

```diff
@@ -122,4 +125,128 @@ public override async Task<string> SaveFileAsync(MemoryStream data, Cancellation
             throw new MinioFilePersistenceException("Failed to upload file to object storage", exception);
         }
     }
+
+    public override async Task<string> SaveJsonFileAsync<T>(T data, string filename, CancellationToken cancellationToken)
+    {
+        ArgumentNullException.ThrowIfNull(data);
+        if (string.IsNullOrEmpty(filename)) throw new ArgumentException("No filename provided to save", nameof(filename));
+
+        // The type of the data that can be dynamically determined and used to deserialize
+        Type dataType = typeof(T);
+        string? objectType = dataType.Name;
+        _logger.LogInformation("The object type of the saved data is {ObjectType}", objectType);
+
+        string createdAt = _clock.GetCurrentInstant().ToDateTimeUtc().ToString(CreatedAtDateTimeFormat);
+
+        // Serialize data to a JSON string
+        string serializedJsonData = JsonSerializer.Serialize(data);
+        // Convert serialized string data into byte stream to save as a file
+        MemoryStream dataJsonStream = new(Encoding.UTF8.GetBytes(serializedJsonData));
```

Comment:
Can you please use the `IMemoryStreamManager` to get a new stream. Also, you can serialize directly into the memory stream.

```
        IMemoryStreamManager memoryStreamManager;
        object data = new object();

        var stream = memoryStreamManager.GetStream();
        System.Text.Json.JsonSerializer.Serialize(stream, data);
```

Reply:
Done.